auth/client/src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -679,6 +702,51 @@ public AuthenticationConfiguration usePassword(String password) {
         return usePassword(password == null ? null : ClearPassword.createRaw(ClearPassword.ALGORITHM_CLEAR, password.toCharArray()));
     }
 
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given masked password to authenticate.
+     *
+     * @param maskedPasswordBytes the masked password bytes (must not be {@code null})
+     * @param algorithm the algorithm (can be {@code null})
+     * @param initialKeyMaterial the initial key material (can be {@code null})
+     * @param iterationCount the iteration count (must not be 0)
+     * @param salt the salt bytes (must not be {@code null})
+     * @param initializationVector the initialization vector (can be {@code null})
+     * @return the new configuration
+     */
+    public AuthenticationConfiguration usePassword(byte[] maskedPasswordBytes, String algorithm, char[] initialKeyMaterial, int iterationCount, byte[] salt, byte[] initializationVector) throws Exception {
+        if (iterationCount == 0) throw log.invalidIterationCount();
+        if (salt == null) throw log.invalidSalt();
+        if (maskedPasswordBytes == null) throw log.invalidMaskedPasswordBytes();
+        if (algorithm == null) algorithm = MaskedPassword.ALGORITHM_MASKED_MD5_DES;
+        if (initialKeyMaterial == null) initialKeyMaterial = "somearbitrarycrazystringthatdoesnotmatter".toCharArray();
+        final MaskedPasswordSpec spec = new MaskedPasswordSpec(initialKeyMaterial, iterationCount, salt, maskedPasswordBytes, initializationVector);
+
+        PasswordFactory factory = PasswordFactory.getInstance(algorithm);
+        MaskedPassword password = factory.generatePassword(spec).castAs(MaskedPassword.class);
+        return useMaskedPassword(password);
+    }
+
+
+    /**
+     * Create a new configuration which is the same as this configuration, but which uses the given masked password to authenticate.
+     *
+     * @param maskedPasswordBytes the masked password bytes (must not be {@code null})
```

Comment:
Maybe this could be renamed to just ```maskedPassword``` and in the description indicate that it's the masked password, as a String.

Reply:
fixed